src/metabase/models/field_values.clj

```diff
@@ -234,14 +235,17 @@
   [field-or-field-id]
   (if-not (map? field-or-field-id)
     (let [field-id (u/the-id field-or-field-id)]
-      (recur (or (t2/select-one ['Field :base_type :visibility_type :has_field_values] :id field-id)
+      (recur (or (t2/select-one 'Field :id field-id)
```

Comment:
To future-proof ourselves a bit I think we should have this method take a MLv2 `:metadata/column` rather than an application database `Field`, after all we are trying to decouple the drivers/QP/MLv2 from the application database completely going forward. 

The main difference is the use of `kebab-case` keys rather than `snake_case` keys.

You have a couple options for making sure you have a `:metabase.lib.schema.metadata/column` here:

1. Re-fetch the MLv2 metadata from the app DB with something like `(t2/select-one :metadata/column (u/the-id field-or-field-id))`
2. Convert the `:model/Field` to a `:metadata/column` using `metabase.lib.metadata.jvm/instance->metadata`. The latter is probably better since you can avoid an extra DB call

Reply:
Here I've tried to use `(t2/select-one :metadata/column (u/the-id field-or-field-id))` which seem to do the right thing. But then, reading through the `column` malli definition, I've got to the sentece: `Note that all metadata providers may not return this column.` I'm not _yet_ able to fully comprehend ramifications of that, hence I decided to return this snippet to its original pre-PR state. My reasoning was something along "maybe with some future change this provider is used here and that key could be missing".